Submitted By: Guy Dalziel <krendoshazin at dementedfury dot org>
Date: 2009-07-15
Initial Package Version: 1.8
Upstream Status: Not submitted (LFS specific)
Origin: Guy Dalziel, Randy McMurchy, and David Jensen
Description: Fixes 'make install' so that all files are installed.

--- libmpeg3-1.8/Makefile	2008-08-07 04:33:09.000000000 +0100
+++ libmpeg3-1.8-patched/Makefile	2009-07-15 21:59:27.726173705 +0100
@@ -1,13 +1,10 @@
 CC = gcc
 NASM = nasm
+VERSION_NAME := $(shell basename $(PWD))
 USE_MMX = 0
 USE_CSS = 1
 A52DIR := $(shell expr a52dec* )
-
-
-ifeq ("$(DST)", "")
-DST=/usr/bin
-endif
+PREFIX=/usr
 
 ifeq ($(origin CFLAGS), environment)
 HAVE_CFLAGS := y
@@ -191,7 +188,13 @@
 		-ldl
 
 install: 
-	cp $(UTILS) $(DST)
+	if [ ! -d $(DESTDIR)$(PREFIX)/bin ]; then install -v -m 755 -d $(DESTDIR)$(PREFIX)/bin; fi
+	if [ ! -d $(DESTDIR)$(PREFIX)/lib ]; then install -v -m 755 -d $(DESTDIR)$(PREFIX)/lib; fi
+	if [ ! -d $(DESTDIR)$(PREFIX)/include ]; then install -v -m 755 -d $(DESTDIR)$(PREFIX)/include; fi
+	install -v -m 755 -t $(DESTDIR)$(PREFIX)/bin $(UTILS)
+	install -v -m 644 -t $(DESTDIR)$(PREFIX)/lib $(OUTPUT)
+	install -v -m 644 -t $(DESTDIR)$(PREFIX)/include libmpeg3.h mpeg3private.h mpeg3protos.h
+	install -v -m 644 -D docs/index.html $(DESTDIR)$(PREFIX)/share/doc/$(VERSION_NAME)/index.html
 
 clean:
 	rm -rf $(OBJDIR)
